src/audit/unsecure_commands_allowed.rs

```diff
@@ -0,0 +1,117 @@
+use crate::audit::WorkflowAudit;
+use crate::finding::{Confidence, Finding, FindingBuilder, Severity, SymbolicLocation};
+use crate::models::{Steps, Workflow};
+use crate::state::AuditState;
+use github_actions_models::common::{Env, EnvValue};
+use github_actions_models::workflow::job::StepBody;
+use github_actions_models::workflow::Job;
+use std::ops::Deref;
+
+static ID: &str = "unsecure-commands-allowed";
+static DESCRIPTION: &str = "execution of insecure workflow commands is enabled";
+
+pub(crate) struct UnsecureCommandsAllowed;
+
+impl UnsecureCommandsAllowed {
+    fn insecure_commands_allowed<'w>(
+        &self,
+        workflow: &'w Workflow,
+        location: SymbolicLocation<'w>,
+    ) -> Finding<'w> {
+        FindingBuilder::new(ID, DESCRIPTION)
+            .confidence(Confidence::High)
+            .severity(Severity::High)
+            .add_location(
+                location
+                    .with_keys(&["env".into()])
+                    .annotated("insecure GHA commands allowed here"),
+            )
+            .build(workflow)
+            .expect("Cannot build a Finding instance")
+    }
+
+    fn has_unsecure_commands_enabled(&self, env: &Env) -> bool {
+        if let Some(value) = env.get("ACTIONS_ALLOW_UNSECURE_COMMANDS") {
+            // Simple emulation for type coercion from YAML declared values at GHA runtime
+            match value {
+                EnvValue::String(string) => !string.is_empty(),
+                EnvValue::Number(number) => *number > 0.0,
+                EnvValue::Boolean(it) => *it,
+            }
```

Comment:
Thanks for the diligence here, but the good news is that GHA is much simpler: they use C#'s `bool.TryParse` which only allows `true` and `false` tokens, case insensitive: https://learn.microsoft.com/en-us/dotnet/api/system.boolean.tryparse?view=net-9.0

So, I think it'd be enough to just check `EnvValue::String` here and treat `true` (case insensitive) as the "true" case :slightly_smiling_face:

Reply:
Source ref: https://github.com/actions/runner/blob/e640a9fef36e174358074a2d7fcc3c91eea67fdf/src/Runner.Worker/ActionCommandManager.cs#L252-L255

Reply:
Improved that [here](https://github.com/woodruffw/zizmor/pull/176/commits/f35682c69214f1f8e067af61d80acd434308f288)